src/features/profile/component/modals/edit-iam-profile-details/edit-iam-profile-details.tsx

```diff
@@ -0,0 +1,255 @@
+    >
+      <DialogHeader>
+        <DialogTitle>Edit profile details</DialogTitle>
+        <DialogDescription>Keep your details accurate and up to date.</DialogDescription>
+      </DialogHeader>
+      <Form {...form}>
+        <form onSubmit={handleSubmit(onSubmit)} className="flex flex-col gap-5">
+          <div className="grid grid-cols-1 sm:grid-cols-2 gap-4">
+            <FormField
+              control={control}
+              name="fullName"
+              rules={{ required: 'Full Name is required' }}
+              render={({ field }) => (
+                <FormItem>
+                  <Label>Full Name*</Label>
+                  <FormControl>
+                    <Input {...field} placeholder="Enter your full name" />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            <FormField
+              control={control}
+              name="role"
+              render={({ field }) => (
+                <FormItem>
+                  <Label>Role</Label>
+                  <Select
+                    onValueChange={field.onChange}
+                    defaultValue={field.value}
+                    value={field.value}
+                  >
+                    <FormControl>
+                      <SelectTrigger>
+                        <SelectValue placeholder="Select a role" />
+                      </SelectTrigger>
+                    </FormControl>
+                    <SelectContent>
+                      {AVAILABLE_ROLES.map((role) => (
+                        <SelectItem key={role} value={role}>
+                          {role.charAt(0).toUpperCase() + role.slice(1)}
+                        </SelectItem>
+                      ))}
+                    </SelectContent>
+                  </Select>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+
+            <FormField
+              control={control}
+              name="email"
+              render={({ field }) => (
+                <FormItem>
+                  <Label>Email</Label>
+                  <FormControl>
+                    <Input {...field} disabled />
+                  </FormControl>
+                </FormItem>
+              )}
+            />
+            <FormField
+              control={control}
+              name="phoneNumber"
+              rules={{
+                validate: (value) => {
+                  if (!value) return 'Phone number is required';
+                  if (!isPossiblePhoneNumber(value)) return 'Phone number length is invalid';
+                  if (!isValidPhoneNumber(value)) return 'Invalid phone number';
+                  return true;
+                },
+              }}
+              render={({ field }) => (
+                <FormItem>
+                  <Label>Mobile No.</Label>
+                  <FormControl>
+                    <PhoneInput
+                      {...field}
```

Comment:
@afiapreety, We're using the same styling for PhoneInput in multiple places, including here and in the Edit Profile modal. Let's refactor by creating a reusable PhoneInput component. Let me do it later